src/group_theory/order_of_element.lean

```diff
@@ -124,21 +124,57 @@ section order_of
 
 section monoid
 variables {α} [monoid α]
+variables {H : Type u} [add_monoid H] {x : H}
 
 open_locale classical
 
+/-- `add_order_of h` is the additive order of the element `x`, i.e. the `n ≥ 1`, s. t. `n •ℕ x = 0`
+if it exists. Otherwise, i.e. if `x` is of infinite order, then `add_order_of x` is `0` by
+convention.-/
+noncomputable def add_order_of (x : H) : ℕ :=
+if h : ∃ n, 0 < n ∧ n •ℕ x = 0 then nat.find h else 0
+
 /-- `order_of a` is the order of the element `a`, i.e. the `n ≥ 1`, s.t. `a ^ n = 1` if it exists.
 Otherwise, i.e. if `a` is of infinite order, then `order_of a` is `0` by convention.-/
 noncomputable def order_of (a : α) : ℕ :=
 if h : ∃ n, 0 < n ∧ a ^ n = 1 then nat.find h else 0
 
+attribute [to_additive add_order_of] order_of
+
+@[simp] lemma add_order_of_of_mul_eq_order_of (a : α) :
+  add_order_of (additive.of_mul a) = order_of a :=
+begin
+  rw add_order_of,
+  rw order_of,
+  split_ifs;
+  refl
+end
+
+@[simp] lemma order_of_of_add_eq_add_order_of (x : H) :
+  order_of (multiplicative.of_add x) = add_order_of x :=
+begin
+  rw add_order_of,
+  rw order_of,
+  split_ifs;
+  refl,
+end
```

Comment:
```suggestion
lemma of_mul_pow {A : Type*} [monoid A] (x : A) (n : ℕ) :
  additive.of_mul (x ^ n) = n •ℕ (additive.of_mul x) :=
(congr_arg additive.of_mul (of_add_nsmul (additive.of_mul x) n)).symm

@[simp] lemma of_mul_eq_zero {A : Type*} [monoid A] {x : A} :
  additive.of_mul x = 0 ↔ x = 1 := iff.rfl

@[simp] lemma of_add_eq_one {A : Type*} [add_monoid A] {x : A} :
  multiplicative.of_add x = 1 ↔ x = 0 := iff.rfl

@[simp] lemma add_order_of_of_mul_eq_order_of (a : α) :
  add_order_of (additive.of_mul a) = order_of a :=
by simp [add_order_of, order_of, ←of_mul_pow]

@[simp] lemma order_of_of_add_eq_add_order_of (x : H) :
  order_of (multiplicative.of_add x) = add_order_of x :=
by simp [add_order_of, order_of, ←of_add_nsmul]
```

Right now `split_ifs` eagerly unifies the two dite conditions because currently, they are defeq. But if `multiplicative` and `additive` weren't as such, then these proofs would be more brittle. By including the lemmas above, you can avoid this `rfl` based step in `split_ifs`. Perhaps some similar lemmas would be helpful for things below.

The `of_mul_pow` and related lemmas should probably go in `algebra/group_power/basic.lean`. I plagiarized them from myself from https://github.com/leanprover-community/mathlib/pull/6773/files#diff-037522102a2edcd0b33801a2dc6dc56ea0eacefd5bd58c17402eec520dff4fffR825-R850

Reply:
Thank you. I'm was actually quite puzzled by how much `split_ifs` deals with.